Ical.Net/DataTypes/IDateTime.cs

```diff
@@ -10,129 +10,133 @@ namespace Ical.Net.DataTypes;
+    /// The value has no associated timezone.<br/>
+    /// The precision of the time part is up to seconds.
+    /// <para/>
+    /// Use <see cref="IsUtc"/> along with <see cref="TzId"/> and <see cref="IsFloating"/>
+    /// to control how this date/time is handled.
     /// </summary>
-    DateTime Value { get; set; }
+    DateTime Value { get; }
 
     /// <summary>
-    /// Gets/sets whether or not this date/time value contains a 'date' part.
+    /// Returns <see langword="true"/>, if the date/time value contains a 'time' part.
     /// </summary>
-    bool HasDate { get; set; }
+    bool HasTime { get; }
 
     /// <summary>
-    /// Gets/sets whether or not this date/time value contains a 'time' part.
+    /// Returns <see langword="true"/>, if the date/time value is floating.
+    /// <para/>
+    /// A floating date/time value does not include a timezone identifier or UTC offset,
+    /// so it is interpreted as local time in the context where it is used.
+    /// <para/>
+    /// A floating date/time value is useful when the exact timezone is not
+    /// known or when the event should be interpreted in the local timezone of
+    /// the user or system processing the calendar data.
     /// </summary>
-    bool HasTime { get; set; }
+    bool IsFloating { get; }
 
     /// <summary>
-    /// Gets/sets the time zone ID for this date/time value.
+    /// Gets the timezone ID that applies to the <see cref="Value"/>.
     /// </summary>
-    string? TzId { get; set; }
+    string? TzId { get; }
 
     /// <summary>
-    /// Gets the year for this date/time value.
+    /// Gets the year that applies to the <see cref="Value"/>.
     /// </summary>
     int Year { get; }
 
     /// <summary>
-    /// Gets the month for this date/time value.
+    /// Gets the month that applies to the <see cref="Value"/>.
     /// </summary>
     int Month { get; }
 
     /// <summary>
-    /// Gets the day for this date/time value.
+    /// Gets the day that applies to the <see cref="Value"/>.
     /// </summary>
     int Day { get; }
 
     /// <summary>
-    /// Gets the hour for this date/time value.
+    /// Gets the hour that applies to the <see cref="Value"/>.
     /// </summary>
     int Hour { get; }
 
     /// <summary>
-    /// Gets the minute for this date/time value.
+    /// Gets the minute that applies to the <see cref="Value"/>.
     /// </summary>
     int Minute { get; }
 
     /// <summary>
-    /// Gets the second for this date/time value.
+    /// Gets the second that applies to the <see cref="Value"/>.
     /// </summary>
     int Second { get; }
 
     /// <summary>
-    /// Gets the millisecond for this date/time value.
+    /// Gets the millisecond that applies to the <see cref="Value"/>.
     /// </summary>
     int Millisecond { get; }
 
     /// <summary>
-    /// Gets the ticks for this date/time value.
+    /// Gets the ticks that applies to the <see cref="Value"/>.
     /// </summary>
     long Ticks { get; }
```

Comment:
The question is applicable to other properties, too: Day, Month, Year...
All can be retrieved from the `Value` property. We could clean-up, then this would bring any advantage. But not in this PR.

Reply:
Sure, my thought on Milliseconds and Ticks is just, that they are truncated starting with this PR, so providing getters for them could be confusing and considered inconsistent.